### PR TITLE
Transfer challenges to the new format

### PR DESCRIPTION
**Pull request by bocharsky-bw**
Partially resolve https://github.com/KnpLabs/KnpUniversity/pull/476#old-challenges

### CODE REVIEW

File: challenges.yml [hunk 1]
```diff
@@ -0,0 +1,184 @@
...
+        eat.php: ~
+        Pizza.php: ~
+    features:
+        editor: ~
+        browser:
+            default_uri: /eat.php
+    grading:
+        check_classes:
+            type: php
+            expect: "(not class_exists('Food\Tasty\Pizza')) and class_exists('Pizza')"
```
Comment (weaverryan) [inline challenges.yml:22]: did you add this `class_exists` to the grader? It's good function btw :+1: I just don't see the commit that adds it

Comment (bocharsky-bw) [inline challenges.yml:22]: Oh, I wanted to add this but forgot. I'm going to add it now.

File: challenges.yml [hunk 2]
```diff
@@ -0,0 +1,184 @@
+addsimplenamespacecoding:
```
Comment (weaverryan) [inline challenges.yml:1]: Ah, this is perfect actually! These keys should be based off of the original class names:
- `Challenges\AddSimpleNamespaceCoding` -> `addsimplenamespacecoding` (exactly as you have)
- (if there is a namespace) `Challenges\Foo\SomeChallenge` -> `foo_somechallenge`

If we do this, then if a user previously answered a challenge, KnpU.com will remember that they got it right (it links the old challenge to the new one).

File: challenges.yml (hunk 1 above)
Comment (bocharsky-bw) [inline challenges.yml:22]: Done in https://github.com/KnpLabs/KnpUniversity/commit/5a6f2568b2ab8ac7b54e189397b321f0649194d9

File: challenges.yml (hunk 2 above)
Comment (bocharsky-bw) [inline challenges.yml:1]: @weaverryan  OK! BTW, `Shift` + `cmd` + `U` shortcut in PhpStorm converts `ClassName` to the `classname`! :)

**Comment by weaverryan** [comment]
Really nice work. So far (some final details might change, but probably not), this looks exactly like it should. If you notice some "features" of the grading system that would be helpful, let's talk about them. For example, I originally allowed you to have multiple `expect` lines per grading, but then removed it since you can use `and` in the expression. But anyways, there may be some things we can change that will help us.

File: challenges.yml (hunk 2 above)
Comment (weaverryan) [inline challenges.yml:1]: wow, good tip :)

**Pull request merged**